### PR TITLE
Replaced Toast with Snackbar in ReadText.kt

### PR DESCRIPTION
**Pull request by Prince-kushwaha**
## Purpose / Description
Replaced Toast with Snackbar in ReadText.kt

## Fixes
Related to #13411 
## How Has This Been Tested?
Tested on Android Device 

## Video

For Initializing TTS
https://github.com/ankidroid/Anki-Android/assets/65972015/8a8fdd8d-616f-4736-a9c7-8c41ca0cf896

For No text-to-speech language available

https://github.com/ankidroid/Anki-Android/assets/65972015/9f5757c1-7e4f-42db-9432-7ef1e3e5d56b

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/ReadText.kt [hunk 1]
```diff
@@ -252,7 +253,7 @@ object ReadText {
                     Timber.d("TTS initialized and available languages found")
                     (context as AbstractFlashcardViewer).ttsInitialized()
                 } else {
-                    showThemedToast(context, context.getString(R.string.no_tts_available_message), false)
+                    ankiActivityContext!!.showSnackbar(R.string.no_tts_available_message)
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/ReadText.kt:256]: This doesn't seem safe. (not related to your PR) Design-wise, Idk even why the need of a activity weakReference or how is this a good idea. Ignoring that, avoid using `!!` here, check if the context isn't null and show a toast if it is

**Review by BrayanDSO — CHANGES_REQUESTED** [review]

**Comment by Prince-kushwaha** [comment]
@david-allison  review this pr

**Review by david-allison — CHANGES_REQUESTED** [review]
I believe Brayan's comment hasn't been addressed. This doesn't show a toast if the cast fails

**Comment by Prince-kushwaha** [comment]
@david-allison  it show the toast in all  scenario. cast only fail when we runing unit test  case and that time not showing Toast.

**Review by mikehardy — APPROVED** [review]
Thank you!

**Pull request merged**